public/components/register_model/tag_field.tsx

```diff
@@ -113,20 +122,50 @@ export const ModelTagField = ({
     },
   });
 
+  const selectedTagGroup = useMemo(
+    () => tagGroups.find((t) => t.name === tagKeyController.field.value),
+    [tagGroups, tagKeyController]
+  );
+
+  const tagTypeController = useController({
+    name: `tags.${index}.type` as const,
+    control,
+  });
+
+  useEffect(() => {
+    if (selectedTagGroup) {
+      tagTypeController.field.onChange(selectedTagGroup.type);
```

Comment:
Could we move this effect to tagKey onChange event handler? According [react's suggestion](https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers).